woocommerce/woocommerce
# Add editable headings for cart and checkout order summary subtotal

### Changes proposed in this Pull Request:

Partially addresses https://github.com/woocommerce/woocommerce/issues/52043

In https://github.com/woocommerce/woocommerce/pull/52892 I implemented editable headings via entities, but overall didn't like the experience or the needless added complexity.

This takes a more traditional route with block attributes and the experience feels much better. Also the last PR was much too large so I've opted to break it into smaller PR's to address each heading one at a time. Easier for reviewing and testing.



### How to test the changes in this Pull Request:

For cart and checkout:

1. Edit the page, find the subtotal in the order summary, click on the text to edit it
2. Smoke test various editing scenarios (empty out all the text, save empty, etc)
3. Save the changes
4. Smoke test the changes display as expected on the frontend.

Test translations:

1. Have translations installed for Nederlands (but stay in English for now)
2. Edit the page find the subtotal, edit it to anything and save
3. Now go back and edit the page again setting the heading to the english default `Subtotal`
4. Now swap to Nederlands language in WordPress
5. Now go to front-end and see that the heading should be translated to `Subtotaal`



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Allow order summary subtotal heading to be edited in both cart and checkout blocks.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
